pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/IndexType.java

```diff
@@ -38,6 +38,11 @@
  * @param <IC> the {@link IndexCreator} subclass that should be used to create indexes of this type.
  */
 public interface IndexType<C extends IndexConfig, IR extends IndexReader, IC extends IndexCreator> {
+  enum IndexBuildLifecycle {
+    DURING_SEGMENT_CREATION,
+    POST_SEGMENT_CREATION,
+    CUSTOM
```

Comment:
Can you add a javadoc explaining the meaning of each literal?

Reply:
nit: just call it`BuildTime` as this enum is defined as part of IndexType